### ISSUE TITLE
Error in solving

### ISSUE THREAD

**Issue by thofma** [oscar-system/Oscar.jl#3092]
Seen at https://github.com/oscar-system/Oscar.jl/actions/runs/7146374939/job/19463925668?pr=3091#step:7:1356

CC: @ederc 

Stacktrace:
```
solving: Error During Test at /home/runner/work/Oscar.jl/Oscar.jl/test/Rings/solving.jl:46
  Test threw exception
  Expression: sols == (real_solutions(I))[1]
  DivideError: integer division error
  Stacktrace:
   [1] _core_msolve(I::AlgebraicSolving.Ideal{QQMPolyRingElem}; initial_hts::Int64, nr_thrds::Int64, max_nr_pairs::Int64, la_option::Int64, info_level::Int64, precision::Int64)
     @ AlgebraicSolving ~/.julia/packages/AlgebraicSolving/3tMWz/src/algorithms/solvers.jl:103
   [2] #real_solutions#34
     @ ~/.julia/packages/AlgebraicSolving/3tMWz/src/algorithms/solvers.jl:392 [inlined]
   [3] #real_solutions#404
     @ ~/work/Oscar.jl/Oscar.jl/src/Rings/solving.jl:43 [inlined]
   [4] real_solutions
     @ ~/work/Oscar.jl/Oscar.jl/src/Rings/solving.jl:41 [inlined]
   [5] macro expansion
     @ ~/work/Oscar.jl/Oscar.jl/test/Rings/solving.jl:46 [inlined]
   [6] macro expansion
     @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
   [7] top-level scope
     @ ~/work/Oscar.jl/Oscar.jl/test/Rings/solving.jl:2
Test Summary: | Pass  Error  Total
solving       |   48      1     49
ERROR: LoadError: LoadError: Some tests did not pass: 48 passed, 0 failed, 1 errored, 0 broken.
in expression starting at /home/runner/work/Oscar.jl/Oscar.jl/test/Rings/solving.jl:1
in expression starting at /home/runner/work/Oscar.jl/Oscar.jl/test/runtests.jl:132
ERROR: LoadError: Package Oscar errored during testing
Stacktrace:
 [1] pkgerror(msg::String)
   @ Pkg.Types /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/Types.jl:55
 [2] test(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; coverage::Bool, julia_args::Cmd, test_args::Cmd, test_fn::Nothing)
   @ Pkg.Operations /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/Operations.jl:1712
 [3] test(ctx::Pkg.Types.Context, pkgs::Vector{Pkg.Types.PackageSpec}; coverage::Bool, test_fn::Nothing, julia_args::Vector{String}, test_args::Cmd, kwargs::Base.Iterators.Pairs{Symbol, IOContext{Base.PipeEndpoint}, Tuple{Symbol}, NamedTuple{(:io,), Tuple{IOContext{Base.PipeEndpoint}}}})
   @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:343
 [4] test(pkgs::Vector{Pkg.Types.PackageSpec}; io::IOContext{Base.PipeEndpoint}, kwargs::Base.Iterators.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:coverage, :julia_args), Tuple{Bool, Vector{String}}}})
   @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:80
 [5] test(; name::Nothing, uuid::Nothing, version::Nothing, url::Nothing, rev::Nothing, path::Nothing, mode::Pkg.Types.PackageMode, subdir::Nothing, kwargs::Base.Iterators.Pairs{Symbol, Any, Tuple{Symbol, Symbol}, NamedTuple{(:coverage, :julia_args), Tuple{Bool, Vector{String}}}})
   @ Pkg.API /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Pkg/src/API.jl:96
 [6] top-level scope
   @ ~/work/_actions/julia-actions/julia-runtest/latest/test_harness.jl:15
 [7] include(fname::String)
   @ Base.MainInclude ./client.jl:444
 [8] top-level scope
   @ none:1
in expression starting at /home/runner/work/_actions/julia-actions/julia-runtest/latest/test_harness.jl:7
Error: Process completed with exit code 1.
```

**Comment by thofma** [comment]
@ederc is the test randomized?

**Comment by ederc** [comment]
No, this test was issue #1743 .

**Comment by thofma** [comment]
Hm, is the implementation randomized?

**Comment by ederc** [comment]
No. Nothing has changed inside msolve for this kind of computation in the last versions. It is also strange that it is happening in Julia 1.6 only and not in PR #3090.

**Comment by thofma** [comment]
I could quickly reproduce it on 1.6 using:
```
julia> using Test; i = 0; while true; @info i; Oscar.test_module("Rings/solving", new = false); i += 1;end
[...]
[ Info: 68
solving: Error During Test at /Users/bla/dev/Oscar/test/Rings/solving.jl:46
  Test threw exception
  Expression: sols == (real_solutions(I))[1]
  DivideError: integer division error
  Stacktrace:
   [1] _core_msolve(I::AlgebraicSolving.Ideal{QQMPolyRingElem}; initial_hts::Int64, nr_thrds::Int64, max_nr_pairs::Int64, la_option::Int64, info_level::Int64, precision::Int64)
```

**Comment by ederc** [comment]
I can confirm this now as a bug in msolve. We will try to provide a new minor release in the next days. You are right (and I was wrong) there is some different randomization for chosing primes which triggers this error sometimes.

**Comment by thofma** [comment]
Thanks for checking

**Comment by ederc** [comment]
Fixed via `msolve_jll v0.6.2+0`

**Comment by lgoettgens** [comment]
The snippet in https://github.com/oscar-system/Oscar.jl/issues/3092#issuecomment-1849472848 doesn't throw an error in the first 300 iterations locally, so I call this resolved.
